src/AppInstallerRepositoryCore/Microsoft/Schema/1_4/DependenciesTable.cpp

```diff
@@ -0,0 +1,375 @@
+
+            SQLite::Builder::StatementBuilder builder;
+            builder.DeleteFrom(s_DependenciesTable_Table_Name).Where(SQLite::RowIDName).In(dependencyRowIds.size());
+
+            SQLite::Statement deleteStmt = builder.Prepare(connection);
+            int bindIndex = 1;
+            for (SQLite::rowid_t rowId : dependencyRowIds)
+            {
+                deleteStmt.Bind(bindIndex, rowId);
+                bindIndex++;
+            }
+
+            deleteStmt.Execute(true);
+
+            savepoint.Commit();
+        }
+
+        void ThrowOnMissingPackageNodes(std::vector<Manifest::Dependency> missingPackageNodes)
+        {
+            if (!missingPackageNodes.empty())
+            {
+                std::string missingPackages{ missingPackageNodes.begin()->Id };
+                std::for_each(
+                    missingPackageNodes.begin() + 1,
+                    missingPackageNodes.end(),
+                    [&](auto& dep) { missingPackages.append(", " + dep.Id);  });
+                THROW_HR_MSG(APPINSTALLER_CLI_ERROR_MISSING_PACKAGE, "Missing packages: %s", missingPackages.c_str());
+            }
+        }
+
+        void InsertManifestDependencies(
+            SQLite::Connection& connection,
+            SQLite::rowid_t manifestRowId,
+            std::set<Manifest::Dependency>& dependencies)
+        {
+            using namespace SQLite::Builder;
+            using namespace Schema::V1_0;
+
+            std::vector<Manifest::Dependency> missingPackageNodes;
+            std::map<Utility::Version, SQLite::rowid_t> versionsMap;
+            std::map<Manifest::Manifest::string_t, SQLite::rowid_t> idsMap;
+            for (const auto& dep : dependencies)
+            {
+                auto depMinVersion = dep.MinVersion.value();
+                auto packageId = dep.Id;
+
+                const auto packageRowId = IdTable::SelectIdByValue(connection, packageId);
+                if (!packageRowId.has_value())
+                {
+                    missingPackageNodes.emplace_back(dep);
+                    continue;
+                }
+
+                auto manifestVersions = ManifestTable::GetAllValuesById<IdTable, VersionTable>(connection, packageRowId.value());
+
+                auto result = std::find_if(
+                    manifestVersions.begin(),
+                    manifestVersions.end(),
+                    [&](auto& manifestVersion)
+                    {
+                        return (std::get<0>(manifestVersion) == depMinVersion.ToString());
+                    }
+                );
+
+                if (result == manifestVersions.end())
+                {
+                    VersionTable::EnsureExists(connection, depMinVersion.ToString());
+                }
+
+                idsMap[packageId] = packageRowId.value();
+                versionsMap[depMinVersion] = VersionTable::SelectIdByValue(connection, depMinVersion.ToString()).value();
```

Comment:
>SelectIdByValue [](http://example.com/codeflow?start=59&length=15)

`EnsureExists` does everything for you, including returning the rowid of the version string.  Line 110 through 124 can be removed and this should just become `VersionTable::EnsureExists(connection, depMinVersion.ToString())`